### PR TITLE
openapi: Validate unregistered type, if they can be found

### PR DESCRIPTION
**Pull request by apelisse**
**What this PR does / why we need it**:
Types that are not registered/hard-coded in kubectl won't be validated, even if they could because they are defined in openapi. If they are neither registered nor in openapi, then skip validation. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes nothing

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/util/openapi/validation/validation.go [hunk 1]
```diff
@@ -76,14 +76,14 @@ func (v *SchemaValidation) validateList(object interface{}) []error {
...
-
 	resource := v.resources.LookupResource(gvk)
 	if resource == nil {
+		if !api.Registry.IsEnabledVersion(gvk.GroupVersion()) {
+			// if we don't have this in our scheme, just skip
+			// validation because its an object we don't recognize
+			return nil
+		}
+
 		return []error{fmt.Errorf("unknown object type %#v", gvk)}
```
Comment (deads2k) [inline pkg/kubectl/cmd/util/openapi/validation/validation.go]: I'm not sure this should be a different case anymore.  If we don't have the information to validate, we can't validate.  I don't know that having a matching type in the scheme ought to matter either way.

Comment (apelisse) [reply pkg/kubectl/cmd/util/openapi/validation/validation.go]: The openapi swagger has everything it needs to validate the payload. And we want to validate custom-types, like service-catalog resources, which are never going to be hard-coded in kubectl.

Comment (apelisse) [reply pkg/kubectl/cmd/util/openapi/validation/validation.go]: @deads2k Can I has lgtm?

Comment (deads2k) [reply pkg/kubectl/cmd/util/openapi/validation/validation.go]: > The openapi swagger has everything it needs to validate the payload. And we want to validate custom-types, like service-catalog resources, which are never going to be hard-coded in kubectl.

I think my comment was unclear.  If we can't find the resource, why don't we always return nil?

Comment (apelisse) [reply pkg/kubectl/cmd/util/openapi/validation/validation.go]: Yeah, we just discussed with @pwittrock. Let's get rid of that condition

**Comment by apelisse** [comment]
@deads2k I fixed the integration test. Actually, s/fixed/removed/ since it was only checking that we failed on known types ... It's not relevant anymore.

**Pull request merged**